tests/Doctrine/Tests/ORM/Functional/PersistentCollectionTest.php

```diff
@@ -95,6 +95,28 @@ public function testMatchingDoesNotModifyTheGivenCriteria()
         $this->assertEmpty($criteria->getMaxResults());
         $this->assertEmpty($criteria->getOrderings());
     }
+
+    public function testPersistClearCollection()
+    {
+        $collectionHolder = new PersistentCollectionHolder();
+        $content = new PersistentCollectionContent('first element');
+        $collectionHolder->addElement($content);
+
+        $this->_em->persist($collectionHolder);
+        $this->_em->flush();
+        $this->_em->clear();
+
+        $collectionHolder = $this->_em->find(PersistentCollectionHolder::class, $collectionHolder->getId());
+        $this->assertEquals(1, $collectionHolder->getCollection()->count());
+        $collectionHolder->getRawCollection()->clear();
+
+        $this->_em->persist($collectionHolder);
```

Comment:
Is this call needed?

Reply:
@Ocramius guess so, so we can check that data was actually removed from db

Reply:
In theory, change tracking should pick this up anyway